src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsReportServiceImpl.java

```diff
@@ -0,0 +1,486 @@
+                                                     null);
+    }
+
+    @Override
+    public PaymentWorksBatchReportVendorItem createBatchReportVendorItem(PaymentWorksVendor pmwVendorWithErrors, List<String> errorMessages) {
+        return new PaymentWorksBatchReportVendorItem(pmwVendorWithErrors.getPmwVendorRequestId(),
+                                                     pmwVendorWithErrors.getProcessTimestamp(),
+                                                     pmwVendorWithErrors.getVendorType(), 
+                                                     pmwVendorWithErrors.getRequestingCompanyLegalName(),
+                                                     pmwVendorWithErrors.getRequestingCompanyLegalFirstName(),
+                                                     pmwVendorWithErrors.getRequestingCompanyLegalLastName(), 
+                                                     pmwVendorWithErrors.getRequestingCompanyTinType(),
+                                                     pmwVendorWithErrors.getRequestingCompanyCorporateEmail(),
+                                                     pmwVendorWithErrors.getInitiatorNetId(),
+                                                     errorMessages);
+    }
+
+    @Override
+    public void generateAndEmailProcessingReport(PaymentWorksNewVendorRequestsBatchReportData reportData) {
+        LOG.debug("generateAndEmailReport entered");
+        File reportFile = generateReport(reportData);
+        sendResultsEmail(reportData, reportFile);
+    }
+    
+    private File generateReport(PaymentWorksNewVendorRequestsBatchReportData reportData) {
+        LOG.debug("generateReport: entered");
+        reportData.populateOutstandingSummaryItemsForReport();
+        initializeReportTitleAndFileName(reportData);
+        writeSummarySubReport(reportData);
+        writeProcessingSubReport(reportData.retrievePaymentWorksVendorsProcessed(), getProcessedSubTitle(), PaymentWorksConstants.PaymentWorksBatchReportMessages.NO_RECORDS_PROCESSED_MESSAGE);
+        writeProcessingSubReport(reportData.retrievePaymentWorksVendorsWithProcessingErrors(), getProcessingErrorsSubTitle(), PaymentWorksConstants.PaymentWorksBatchReportMessages.NO_RECORDS_WITH_VALIDATION_ERRORS_MESSAGE);
+        writeUnprocessedSubReport(reportData.retrieveUnprocessablePaymentWorksVendors(), getUnprocessedSubTitle(), PaymentWorksConstants.PaymentWorksBatchReportMessages.MANUAL_DATA_ENTRY_NOT_REQUIRED);
+        finalizeReport();
+        return getReportWriterService().getReportFile();
+    }
+
+    private void sendResultsEmail(PaymentWorksEmailableReportData reportData, File reportFile) {
+        LOG.info("sendResultsEmail: entered");
+        String body = readReportFileToString(reportData, reportFile);
+        String subject = buildEmailSubject(reportData);
+        if (StringUtils.isNotBlank(getToAddress()) && StringUtils.isNotBlank(getFromAddress()) && StringUtils.isNotBlank(body) && StringUtils.isNotBlank(subject)) {
```

Comment:
It would probably be better to do all these checks and logging inside the sendEmail method.

Reply:
Moved as requested.